python/cuml/_common/handle.pyx

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+
+cimport cuml._common.cuda
+
+
+cdef extern from "cuML.hpp" namespace "ML" nogil:
+    cdef cppclass cumlHandle:
+        cumlHandle() except +
+        void setStream(cuml._common.cuda._Stream s)
+
+
+# TODO: name this properly!
+# TODO: add support for setting custom memory allocators
```

Comment:
For an initial implementation for this it might be sufficient to provide a method to use RMM as device allocator:
```python
def enableRMM(self)
```
the implementation would then call a free C++ function like this:

```cpp
void enableRMM( ML::cumlHandle& h )
{
    std::shared_ptr<ML::rmmAllocatorAdapter> allocator( new ML::rmmAllocatorAdapter() );
    h.setDeviceAllocator( allocator );
}
```

Reply:
My latest commit to this PR enables RMM similar to this proposal. Over and above this code, I only made sure that this helper method is in the cython world to limit the default set of dependencies for libcuml.so.